compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2120,14 +2120,18 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
         case1
       }
       .asInstanceOf[List[CaseDef]]
-    assignType(cpy.Match(tree)(sel, cases1), sel, cases1).cast(pt)
+    var nni = sel.notNullInfo
+    if(cases1.nonEmpty) nni = nni.seq(cases1.map(_.notNullInfo).reduce(_.alt(_)))
```

Comment:
Is there a style guide for `if then`? That avoids the faux pas of making if look like an application.